pkg/reconciler/kuberneteseventsource/kuberneteseventsource.go

```diff
@@ -18,21 +18,91 @@ package kuberneteseventsource
 
 import (
 	"context"
+	"fmt"
+	"os"
 
 	sourcesv1alpha1 "github.com/knative/eventing-sources/pkg/apis/sources/v1alpha1"
-	"github.com/knative/eventing-sources/pkg/controller/kuberneteseventsource/resources"
+	"github.com/knative/eventing-sources/pkg/reconciler/kuberneteseventsource/resources"
 	"github.com/knative/pkg/logging"
 	"go.uber.org/zap"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
+const (
+	raImageEnvVar       = "K8S_RA_IMAGE"
+	controllerAgentName = "kuberneteseventsource-controller"
+)
+
+// reconciler reconciles a KubernetesEventSource object
+type reconciler struct {
+	client.Client
+	recorder            record.EventRecorder
+	scheme              *runtime.Scheme
+	receiveAdapterImage string
+}
+
+var _ reconcile.Reconciler = &reconciler{}
+
+// Add creates a new KubernetesEventSource Controller and adds it to the Manager
+// with default RBAC. The Manager will set fields on the Controller and Start it
+// when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	receiveAdapterImage, defined := os.LookupEnv(raImageEnvVar)
+	if !defined {
+		return fmt.Errorf("required environment variable %q not defined", raImageEnvVar)
+	}
+
+	return add(mgr, newReconciler(mgr, receiveAdapterImage))
+}
+
+func newReconciler(mgr manager.Manager, receiveAdapterImage string) reconcile.Reconciler {
+	return &reconciler{
+		Client:              mgr.GetClient(),
+		scheme:              mgr.GetScheme(),
+		recorder:            mgr.GetRecorder(controllerAgentName),
+		receiveAdapterImage: receiveAdapterImage,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
```

Comment:
Maybe say something here that r will reconcile based on changes to owned ContainerSource as well as KubernetesEventSource.

Reply:
Add clarity to the comment for kuberneteseventsource.add()

Reply:
https://github.com/knative/eventing-sources/issues/204